gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTask.java

```diff
@@ -89,19 +97,35 @@ public GobblinHelixTask(TaskRunnerSuiteBase.Builder builder,
                              builder.getAppWorkPath(),
                              this.jobId);
 
+    Config dynamicConfig = builder.getDynamicConfig()
+        .withValue(GobblinClusterConfigurationKeys.TASK_RUNNER_HOST_NAME_KEY, ConfigValueFactory.fromAnyRef(builder.getHostName()))
+        .withValue(GobblinClusterConfigurationKeys.CONTAINER_ID_KEY, ConfigValueFactory.fromAnyRef(builder.getContainerId()))
+        .withValue(GobblinClusterConfigurationKeys.HELIX_INSTANCE_NAME_KEY, ConfigValueFactory.fromAnyRef(builder.getInstanceName()))
+        .withValue(GobblinClusterConfigurationKeys.HELIX_JOB_ID_KEY, ConfigValueFactory.fromAnyRef(this.helixJobId))
+        .withValue(GobblinClusterConfigurationKeys.HELIX_TASK_ID_KEY, ConfigValueFactory.fromAnyRef(this.helixTaskId));
+
+    Integer partitionNum = getPartitionForHelixTask(taskDriver);
+
+    if (partitionNum == null) {
+      throw new IllegalStateException(String.format("Task %s, job %s on instance %s has no partition assigned",
+          this.helixTaskId, builder.getInstanceName(), this.helixJobId));
+    }
+
+    dynamicConfig = dynamicConfig.withValue(GobblinClusterConfigurationKeys.HELIX_PARTITION_ID_KEY, ConfigValueFactory.fromAnyRef(getPartitionForHelixTask(taskDriver)));
```

Comment:
Instead of calling `getPartitionForHelixTask ` again, we can use `partitionNum ` directly

Reply:
Done. Thanks!